### PR TITLE
Feature: Key info 

### LINKED ISSUE #290: Feature: Read the public information Key
### What do you want to share with us?

The key owner can use the `rstuf` CLI to get public information about the key, the `keyid`, `key type`, and `public key`.
It can be used for [distributed asynchronous signing of the metadata](support distributed async threshold signing).

`rstuf key info`
Ask the user to load the key, same as in the Ceremony or Metadata Update
Prints the key information

```shell
rstuf key info
....
keyid: <hash>
public key: <hash>
key type: <type>
```

Nice to have: An extra parameter to show the private key, but not default (to avoid misusage)

`rstuf key info --private`
### References

- https://github.com/repository-service-tuf/repository-service-tuf/issues/327

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kairoaraujo**
It implements the feature `rstuf key info`.

The command asks the user to load a key and show information about the
key, this information is the secure systems lib key id, the key type,
public key hash and, optionally, the private key hash.

* Make `_load_key` a public function (`load_key`), not part of the ceremony only.

* Fix documentation

Closes #290

### CODE REVIEW

File: repository_service_tuf/cli/key/info.py
Comment (MVrachev) [inline repository_service_tuf/cli/key/info.py:42]: I think we should combine these questions in a helper function given that I use the same question in my pr #284.

**Review by MVrachev — COMMENTED** [review]
I will review it once we merge #284, but we can compare those prs and see common things.
An example is the comment I left, but also the pr you made moving `load_key` to become a helper function.

File: repository_service_tuf/cli/key/info.py
Comment (KAUTH) [inline repository_service_tuf/cli/key/info.py:37]: At least for the `keytype` (you can actually also get key id and public key, not private), I believe we can improve the user experience a bit and infer it from the public key file. Something like:
```suggestion
    filepath: str = prompt.Prompt.ask("Enter the key's [green]filename[/]")
    
    # TODO: Move imports to the top of the file
    import json
    import os
    # If the public key file exists, infer the key type from there.
    public_key_filename = filepath + ".pub"
    if os.path.isfile(public_key_filename) is True:
        with open(public_key_filename, mode="rb") as public_key_file:
            public_key: dict = json.load(public_key_file)
            keytype: str = public_key["keytype"]
    else:
        keytype: str = prompt.Prompt.ask(
            "\nChoose key type",
            choices=KeyType.get_all_members(),
            default=KeyType.KEY_TYPE_ED25519.value,
        )
```

**Review by KAUTH — CHANGES_REQUESTED** [review]
Very nice, I left a few nits and some suggestions.

File: repository_service_tuf/helpers/tuf.py
Comment (lukpueh) [reply repository_service_tuf/helpers/tuf.py:231]: or even simpler... :P

```suggestion
        name = name or key["keyid"][:7]
```

Comment (lukpueh) [reply repository_service_tuf/helpers/tuf.py:231]: That said, I think the most readable form is:

```suggestion
        if not name:
           name = key["keyid"][:7]
```

File: repository_service_tuf/cli/key/info.py
Comment (lukpueh) [reply repository_service_tuf/cli/key/info.py:37]: Unfortunately, we can't because rsa public keys created with the `rstuf key generate` command are stored as PEM, and not in the proprietary json format. See docstrings of used [`securesystemslib.interface.generate_and_write_*_keypair` functions](https://github.com/secure-systems-lab/securesystemslib/blob/main/securesystemslib/interface.py) for reference.

FYI: `securesystemslib.interface` is currently being re-designed. Until that I suggest to use it as is, without too much UX optimisation.

File: docs/source/devel/repository_service_tuf.cli.admin.rst
Comment (lukpueh) [inline docs/source/devel/repository_service_tuf.cli.admin.rst:21]: Is this an unrelated change?

File: docs/source/devel/repository_service_tuf.rst
Comment (lukpueh) [inline docs/source/devel/repository_service_tuf.rst:23]: Is this an unrelated change?

File: repository_service_tuf/cli/key/info.py
Comment (lukpueh) [inline repository_service_tuf/cli/key/info.py:58]: nit: Why `if show_private is True` and `if show_private`? Wouldn't the latter be enough, e.g.: 
```suggestion
    if show_private:
        key_table.add_column(
            "Private Key", justify="center", style="red", no_wrap=True
        )
```

**Review by lukpueh — APPROVED** [review]
Didn't check if the rename/refactor is complete and only skimmed tests, but the code LGTM modulo some style nits, did you can feel free to ignore.

Regardless, I was wondering if `--show-private` is useful. The related issue says "nice to have", but I wasn't sure why. :)

File: repository_service_tuf/cli/key/info.py
Comment (KAUTH) [reply repository_service_tuf/cli/key/info.py:37]: Are you sure?
I tested yesterday with `rstuf key generate` and it created the .pub and private files. Then if the public file exists, it's simply a JSON structure that can be read.
I also tested the whole flow and it worked.

Comment (lukpueh) [reply repository_service_tuf/cli/key/info.py:37]: From looking at the code, I'm pretty sure it doesn't work with RSA keys. But I didn't try it out. Did you test with RSA keys?

Comment (lukpueh) [reply repository_service_tuf/cli/key/info.py:37]: Yep, just tried with rsa. The pub key is in PEM format. So your proposal doesn't work. You could of course try to json decode, and, if that fails, try to load as rsa. But as I said above, I wouldn't spend too much time on optimising the UX around `securesystemslib.interface`, as we're hopefully gonna have something better soon.

Comment (kairoaraujo) [reply repository_service_tuf/cli/key/info.py:37]: Thank you folks!

File: docs/source/devel/repository_service_tuf.cli.admin.rst
Comment (kairoaraujo) [reply docs/source/devel/repository_service_tuf.cli.admin.rst:21]: It is related to the doc build.
It wasn't automatically updating the implementation docs.

File: repository_service_tuf/helpers/tuf.py
Comment (kairoaraujo) [reply repository_service_tuf/helpers/tuf.py]: We no longer update `repository_service_tuf/helpers/tuf.py`

File: repository_service_tuf/helpers/tuf.py
Comment (kairoaraujo) [reply repository_service_tuf/helpers/tuf.py]: We no longer update `repository_service_tuf/helpers/tuf.py`

**Comment by kairoaraujo** [comment]
Hi folks,

I rebased and addressed all the comments.
@lukpueh, I removed the private key information as we don't have a use case.

File: repository_service_tuf/cli/key/info.py
Comment (KAUTH) [inline repository_service_tuf/cli/key/info.py:21]: Nit:
```suggestion
    filepath: str = prompt.Prompt.ask("Enter the private key's [green]filename[/]")
```

**Review by KAUTH — APPROVED** [review]
LGTM! I only left one suggestion regarding the prompt to enter the filename.

**Pull request merged**
